### PR TITLE
Fix to issue #3734 - a block with jump kind BBJ_RETURN could also be empty if the method's return type is TYP_VOID

### LINKED ISSUE #3734: Assertion failures crossgenning some libraries CoreFX under JitStress=1/2
This happens on Windows and from the very brief investigation I have done it appears to be related to the oneReturn changes done few weeks ago. As I said the analysis I performed was really brief, so if incorrect, please add comments.

The assertion is in morph.cpp, line 14177.

Crossgenning the latest versions of the following libraries triggers the following assertion failure:
1. For NETAll, ASP.NET Core Application - HelloWorld, ASP.NET Core MVC Application (the CoreFX libraries)..
    - With JitStress=1 - System.Linq.Expressions.dll, method System.Linq.Expressions.Interpreter.LightCompiler:CompileUnaryExpression(ref):this
    - With JitStress=2 System.Xml.ReaderWriter.dll, method System.Xml.DtdParser:ParseExternalId(int,int,byref,byref):this

Same assertion triggers when crossgenning the following Roslyn library with JitStress=1 the Microsoft.CodeAnalysis.VisualBasic.dll in method Microsoft.CodeAnalysis.VisulBasic.DataFlowPass:CheckedAssignedFromArgumentWrite.

### PR DESCRIPTION
**Pull request by sivarv**
The noway assert that failed is that a basic block with jump kind BBJ_RETURN will have at least one stmnt.  But it turns out that it may not always be the case.  For example, a method returning TYP_VOID could have an empty block with jumpkind BBJ_RETURN.  Such a block could materialize as part of in-lining of methods.

For the repro case, the method being jit compiled has TYP_VOID return.  One of the switch blocks has a call followed by return.  This would result in a basic block that has one stmnt (GT_CALL) and the basic block's jump kind will be BBJ_RETURN.  Later the call gets in-lined and as part of that a separate empty block with jump kind BBJ_RETURN materializes in IR.  This results in an assert failure in fgMorphBlocks().

Fix:
Adjusted noway asserts since a block with jump kind BBJ_RETURN could be empty. 

Fixes #3734

### CODE REVIEW

**Comment by sivarv** [comment]
@dotnet/jit-contrib - please review this.

File: src/jit/morph.cpp [hunk 1]
```diff
@@ -14165,28 +14165,32 @@ void                Compiler::fgMorphBlocks()
                     fgReturnCount--;
                 }
 
-                // Block is guaranteed to have last stmt as its jump kind is BBJ_RETURN.
-                // Note that it a block with jump kind BBJ_RETURN not necessarily needs to end with GT_RETURN.
+                // Note 1: A block is not guaranteed to have a last stmt if its jump kind is BBJ_RETURN.
+                // For example a method returning void could have an emtpy block with jump kind BBJ_RETURN.
```
Comment (BruceForstall) [inline src/jit/morph.cpp]: typo: emtpy=>empty

File: src/jit/morph.cpp [hunk 2]
```diff
@@ -14165,28 +14165,32 @@ void                Compiler::fgMorphBlocks()
...
+                // Such blocks do materialize as part of in-lining.
+                //
+                // Note 2: A block with jump kind BBJ_RETURN not necessarily needs to end with GT_RETURN.
```
Comment (BruceForstall) [inline src/jit/morph.cpp]: Grammar is awkward. Consider "Note 2: A block with jump kind BBJ_RETURN does not necessarily need to end with GT_RETURN."

File: src/jit/morph.cpp [hunk 3]
```diff
@@ -14197,8 +14201,13 @@ void                Compiler::fgMorphBlocks()
                     //make sure that copy-prop ignores this assignment.
                     last->gtStmt.gtStmtExpr->gtFlags |= GTF_DONT_CSE;
                 }
-                else if (ret->OperGet() == GT_RETURN)
+                else if (ret != nullptr && ret->OperGet() == GT_RETURN)
                 {
+                    // This block must be ending with a GT_RETURN
```
Comment (BruceForstall) [inline src/jit/morph.cpp]: Suggest rewording: "// This block ends with a GT_RETURN"

**Comment by BruceForstall** [comment]
Do you have a unit test you could add?

**Comment by BruceForstall** [comment]
LGTM

**Comment by LLITCHEV** [comment]
Looks fine, assuming all the test jobs pass.

File: src/jit/morph.cpp (hunk 1 above)
Comment (sivarv) [inline src/jit/morph.cpp]: Done.

File: src/jit/morph.cpp (hunk 2 above)
Comment (sivarv) [inline src/jit/morph.cpp]: Done

File: src/jit/morph.cpp (hunk 3 above)
Comment (sivarv) [inline src/jit/morph.cpp]: Done.

**Comment by sivarv** [comment]
@BruceForstall - this happens while crossgen'ing CoreFx/roslyn assemblies under JitStress=1/2. I think we will soon be automating crossgening of corefx/roslyn/asp.net assemblies as part of OSS CI runs.

**Comment by sivarv** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test
